dojo/unittests/test_jira_import_and_pushing_api.py

```diff
@@ -223,6 +223,15 @@ def test_import_twice_push_to_jira_push_all_issues(self):
         # duplicates shouldn't be sent to JIRA
         self.assert_jira_issue_count_in_test(test_id1, 0)
 
+    def test_engagement_epic_mapping(self):
+        eng = self.get_engagement(3)
+        self.create_engagement_epic(eng)
+        import0 = self.import_scan_with_params(self.zap_sample5_filename, push_to_jira=True, engagement=3)
+        test_id = import0['test']
+        self.assert_jira_issue_count_in_test(test_id, 2)
+        self.assert_epic_issue_count(eng, 2)
+        self.assert_cassette_played()
+
```

Comment:
I think they main problem we saw in the past that if engagement mapping was enabled, the epic was not created and then pushing failed. So I think the tests should test the full flow including creation of the epic. And also the case where findings are pushed and epic mapping is enabled but the epic is not present, it should create a new epic maybe or if not, we should test for the epic not being created in that case.

Reply:
The full flow of the epic being created is present here. Specifically on line 228. I don't like the case of auto creating epics for mapping if the epic did not already exist. This means it would have to be explicitly decided to make it map by using the "creat epic" button on the new/edit engagement page. There is error handling present for this exact use case, but I guess a test couldn't hurt. 

So just to be clear. You want a test to show epic mapping enabled, epic is not created, findings are still pushed.

Reply:
In theory if the epic mapping is enabled, the epic is created. But sometimes this fails, for example if the jira instance requires some extra fields which dojo doesn't know/provide. So for the scenario we should verify that the findings can still be pushed. And if we see it as a requirement that dojo should not try to create the epic, we should assert that as well.

Reply:
No that's not how that works. That would be fine if the jira is created on the engagement level,  but what if it is inherited from the product? Should every engagement have an epic or just the ones you explicitly ask for?